irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -106,6 +107,35 @@ OnDemandOrderingServiceImpl::onRequestProposal(consensus::Round round) {
 
 // ---------------------------------| Private |---------------------------------
 
+/**
+ * Get transactions from the given batches queue. Does not break batches -
+ * continues getting all the transactions from the ongoing batch until the
+ * required amount is collected.
+ * @param requested_tx_amount - amount of transactions to get
+ * @param tx_batches_queue - the queue to get transactions from
+ * @return transactions
+ */
+static std::vector<std::shared_ptr<shared_model::interface::Transaction>>
+getTransactions(size_t requested_tx_amount,
+                tbb::concurrent_queue<TransactionBatchType> &tx_batches_queue) {
+  TransactionBatchType batch;
+  std::vector<std::shared_ptr<shared_model::interface::Transaction>> collection;
+  std::unordered_set<std::string> inserted;
```

Comment:
Why do you need this set?

Reply:
It is there to avoid hash duplicates. It was there originally: https://github.com/hyperledger/iroha/blob/0c373792a7a21afca6f5644013457238af01b43f/irohad/ordering/impl/on_demand_ordering_service_impl.cpp#L187